libsolidity/experimental/analysis/TypeInference.cpp

```diff
@@ -776,8 +792,14 @@ bool TypeInference::visit(TypeDefinition const& _typeDefinition)
 
 	std::vector<Type> arguments;
 	if (_typeDefinition.arguments())
-		for (size_t i = 0; i < _typeDefinition.arguments()->parameters().size(); ++i)
-			arguments.emplace_back(m_typeSystem.freshTypeVariable({}));
+		for (ASTPointer<VariableDeclaration> argumentDeclaration: _typeDefinition.arguments()->parameters())
+		{
+			solAssert(argumentDeclaration);
+			Type typeVar = typeAnnotation(*argumentDeclaration);
+			solAssert(std::holds_alternative<TypeVariable>(typeVar));
+			arguments.emplace_back(typeVar);
+		}
+	m_env->fixTypeVars(arguments);
```

Comment:
I just noticed that I could be fixing the variables `TypeInference::visit(VariableDeclaration)` instead. This would automatically make all parameter lists in type context contain fixed variables. Need to try that.

Reply:
I think I'll try that in a separate PR.

But while looking at it, I realized that this would also fix variables in the type used in a class instantiation and I was not fixing them yet. So I added `m_env->fixTypeVars(arguments)` for them.